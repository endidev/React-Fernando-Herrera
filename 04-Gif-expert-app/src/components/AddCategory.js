import React, {useState} from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('')

    const handleInputChange = (e) => {
        setInputValue( e.target.value )
    }

    const handdleSubmit = (e) => {
        e.preventDefault();
        
        if(inputValue.trim().length > 2){
            setCategories( cats => [inputValue, ...cats])
        }

    }

    return (
        <form onSubmit={handdleSubmit}>
        
            <input 
                type='text'
                value= {inputValue}
                onChange={ handleInputChange }
            />
        </form>
    )
}


AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
