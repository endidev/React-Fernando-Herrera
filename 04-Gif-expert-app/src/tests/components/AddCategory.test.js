import '@testing-library/jest-dom';
import { shallow } from "enzyme"
import { AddCategory } from "../../components/AddCategory";


describe('Pruebas en <AddCategory />', () => {

    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={ setCategories }/>);
    
    beforeEach( () => {
        jest.clearAllMocks();

        wrapper = shallow( <AddCategory setCategories={ setCategories }/>);
    })

    test('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    test('Debe cambiar caja de texto', () => {
        
        const input = wrapper.find('input');

        const value = 'Hola mundo';
        input.simulate('change', { target: { value } });
        
        expect(wrapper.find('p').text().trim()).toBe(value);
    })

    test('No debe postear informacion con submit', () => {

        wrapper.find('form').simulate('submit', { preventDefault(){} })

        expect( setCategories ).not.toHaveBeenCalled();
    })
    
    test('debe de llamar el setCategories, y limpiar la caja de texto', () => {
        
        const value = 'Yu yu hakusho';

        wrapper.find('input').simulate('change', { target: { value }});


        wrapper.find('form').simulate('submit', { preventDefault(){} })

        expect( setCategories ).toHaveBeenCalled();
        expect( setCategories ).toHaveBeenCalledTimes(1);
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function));


        // expect( inputValue ).toBe( '' );
        expect( wrapper.find('input').prop('value')).toBe('');
    })
    
})