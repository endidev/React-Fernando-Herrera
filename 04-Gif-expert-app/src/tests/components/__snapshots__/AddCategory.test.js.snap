// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> Debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
